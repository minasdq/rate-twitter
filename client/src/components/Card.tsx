import {
  Card as MuiCard, CardContent, CardMedia, Theme,
} from '@mui/material';
import { makeStyles } from 'tss-react/mui';

interface CardProps {
  content: string,
  media: JSX.Element
}

const useStyles = makeStyles()((theme: Theme) => ({
  cardContainer: {
    margin: theme.spacing(3, 3, 0, 0),
    width: 200,
    height: 200,
  },
  media: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    margin: theme.spacing(3, 'auto'),
  },
  content: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
}));

const Card = ({
  content, media,
}: CardProps) => {
  const { classes } = useStyles();

  return (
    <MuiCard className={classes.cardContainer}>
      <CardMedia component="div" className={classes.media}>
        {media}
      </CardMedia>
      <CardContent className={classes.content}>
        {content}
      </CardContent>
    </MuiCard>
  );
};

export default Card;
