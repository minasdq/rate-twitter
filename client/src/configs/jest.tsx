// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';

import { createTheme, ThemeProvider } from '@mui/material/styles';

// eslint-disable-next-line import/no-extraneous-dependencies
import '@testing-library/jest-dom/extend-expect';

const theme = createTheme();

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  }
  )),
});

const customizedRender = (
  element: React.ReactElement,
) => {
  const { rerender, ...rest } = render(
    <RecoilRoot>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </RecoilRoot>,
  );

  return {
    ...rest,
    rerender(innerElement = element) {
      return rerender(
        <RecoilRoot>
          <ThemeProvider theme={theme}>
            { innerElement }
          </ThemeProvider>
        </RecoilRoot>,
      );
    },
  };
};

// eslint-disable-next-line import/prefer-default-export
export { customizedRender as render };
