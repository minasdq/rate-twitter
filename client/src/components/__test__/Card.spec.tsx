import { HeartIcon } from '@heroicons/react/outline';

import Card from '../Card';

import { render } from 'Configs/jest';

it('it should match snapshot', () => {
  const { container } = render(
    <Card media={<HeartIcon />} content="4" />,
  );

  expect(container).toMatchSnapshot();
});
