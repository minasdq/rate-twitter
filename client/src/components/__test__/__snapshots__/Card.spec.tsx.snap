// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-t1sepn-MuiPaper-root-MuiCard-root-cardContainer"
  >
    <div
      class="MuiCardMedia-root css-1mv33r3-MuiCardMedia-root-media"
    >
      <svg
        aria-hidden="true"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
    <div
      class="MuiCardContent-root css-3dotsu-MuiCardContent-root-content"
    >
      4
    </div>
  </div>
</div>
`;
